presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixTableProperties.java

```diff
@@ -160,86 +158,70 @@ public PhoenixTableProperties()
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<Boolean> getImmutableRows(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         Boolean value = (Boolean) tableProperties.get(IMMUTABLE_ROWS);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<String> getDefaultColumnFamily(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         String value = (String) tableProperties.get(DEFAULT_COLUMN_FAMILY);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<String> getBloomfilter(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         String value = (String) tableProperties.get(BLOOMFILTER);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<Integer> getVersions(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         Integer value = (Integer) tableProperties.get(VERSIONS);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<Integer> getMinVersions(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         Integer value = (Integer) tableProperties.get(MIN_VERSIONS);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<String> getCompression(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         String value = (String) tableProperties.get(COMPRESSION);
-        if (value == null) {
-            return Optional.empty();
-        }
-        return Optional.of(value);
+        return Optional.ofNullable(value);
+    }
+
+    public static Optional<String> getDataBlockEncoding(Map<String, Object> tableProperties)
+    {
+        requireNonNull(tableProperties);
+
+        String value = (String) tableProperties.get(DATA_BLOCK_ENCODING);
+        return Optional.ofNullable(value);
     }
 
     public static Optional<Integer> getTimeToLive(Map<String, Object> tableProperties)
     {
         requireNonNull(tableProperties);
 
         Integer value = (Integer) tableProperties.get(TTL);
-        if (value == null) {
```

Comment:
Can we capture them in a separate commit ?

Reply:
Then let's do separate PRs. I'll undo this change, and just make sure the new code uses ofNullable.
What about the documentation fixes. Maybe it's best to separate the fix for the Bloomfilter doc as well.